runtime/wasm/Cargo.toml

```diff
@@ -23,3 +23,5 @@ anyhow = "1.0"
 wasmtime = "0.27.0"
 defer = "0.1"
 never = "0.1"
+serde = "1.0.114"
+serde_json = "1.0.56"
```

Comment:
So, I think we can go without `serde_json`, but when it comes to `serde` the `#[derive(Serialize)]` macro is causing problems, since in its expansion there is the statement: `extern crate serde as _serde;` which to my knowledge requires `serde` to be a dependency for the crate (although my knowledge is limited, since I am pretty new to rust hah). Do you have any idea?

A work around might be to implement the `serde::Serialize` trait by hand for `Query` and `Variables` in *debug_tool.rs*, which will make the code a bit more verbose, but it will eliminate the overlapping dependency on serde.

Yet another solution will be to change the macro implementation in `serde_derive`, but I'm not sure if that's an option.